src/jit/codegenarmarch.cpp

```diff
@@ -284,15 +284,21 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 genConsumeAddress(addrNode);
                 addrReg = addrNode->gtRegNum;
 
+                // If addrReg equal to loReg, swap(loReg, hiReg)
+                // This reduces code complexity by only supporting one addrReg overwrite case
+                if (loReg == addrReg)
+                {
+                    loReg = hiReg;
```

Comment:
I don't think that you can do it this way.

I there is a conflict you have to reverse the two loads, not swap the registers used.
Loading the high part (offet+8) into the loReg isn't going to work right

I think with this change you'll end up with the wrong values in loReg and hiReg.